gpkit/constraints/base.py

```diff
@@ -0,0 +1,222 @@
+"Implements Constraint"
+from ..small_classes import HashVector
+from ..keydict import KeySet, KeyDict
+from ..small_scripts import try_str_without
+from ..repr_conventions import _str, _repr, _repr_latex_
+
+
+class Constraint(list):
```

Comment:
The name `Constraint` seems a bit too general/broad. Below is a list of names I like better, in rough order of preference. Do you like any of these?

`RecursiveConstraintContainer`
`ConstraintContainer`
`BaseConstraint` (potentially confusing because 'base' sounds like base case)

... and now I'm wondering: why can't we just combine this with `ConstraintSet`? (sorry if this is a repeat question)